packages/graphback-datasync/src/DataSyncPlugin.ts

```diff
@@ -149,21 +115,74 @@ export class DataSyncPlugin extends GraphbackPlugin {
     }
   }
 
-  protected addDataSyncMetadataFields(schemaComposer: SchemaComposer<any>, model: ModelDefinition) {
+  protected addDataSyncFieldsToModel(schemaComposer: SchemaComposer<any>, model: ModelDefinition) {
     const name = model.graphqlType.name;
     const modelTC = schemaComposer.getOTC(name);
-    const desc = model.graphqlType.description;
-    if (parseMetadata("datasync", desc)) {
-      const metadataFields = createVersionedFields();
-      // metadata fields needed for @versioned
-
-      modelTC.addFields(metadataFields);
 
-      const inputType = schemaComposer.getITC(getInputTypeName(model.graphqlType.name, GraphbackOperationType.FIND))
-      if (inputType) {
-        const metadataInputFields = createVersionedInputFields();
-        inputType.addFields(metadataInputFields);
+    modelTC.addFields({
+      [DataSyncFieldNames.lastUpdatedAt]: {
+        type: GraphbackTimestamp
```

Comment:
This is good to see and kind goes in the way of fixing https://github.com/aerogear/graphback/issues/1349. I've a few questions regarding this line:

What happens if the schema contains `scalar GraphbackTimestamp` already?  Will this work?

Reply:
Thanks for the catch! I haven't fiddled around with scalars yet, hence :P, working on this

Reply:
I have updated this: Now it uses the Graphback scalar if schema already contains it, wondering if this is unnecessary or some other approach is required here:
https://github.com/aerogear/graphback/blob/c02c4b50159a1365d2fe0060435e09a76a50f507/packages/graphback-datasync/src/DataSyncPlugin.ts#L31-L33

Reply:
> I have updated this: Now it uses the Graphback scalar if schema already contains it

Great, this should work. Thanks.  I was wondering if we can add a test case (snaphot test) for this?

> ..., wondering if this is unnecessary or some other approach is required here:
> https://github.com/aerogear/graphback/blob/c02c4b50159a1365d2fe0060435e09a76a50f507/packages/graphback-datasync/src/DataSyncPlugin.ts#L31-L33

I've added a comment in this regard.

Reply:
> Great, this should work. Thanks. I was wondering if we can add a test case (snapshot test) for this?

Thanks for the review! added snapshot now